sql/catalyst/src/test/scala/org/apache/spark/sql/types/DataTypeSuite.scala

```diff
@@ -122,7 +122,7 @@ class DataTypeSuite extends SparkFunSuite {
     val right = StructType(List())
     val merged = left.merge(right)
 
-    assert(merged === left)
+    assert(DataType.equalsIgnoreCompatibleNullability(merged, left))
```

Comment:
I guess we resort to `equalsIgnoreCompatibleNullability` because extra metadata in `merged`? Can we also add assertion for added metadata instead of working around it using `equalsIgnoreCompatibleNullability`?